### PR TITLE
Add one-sample tests + minor changes

### PR DESCRIPTION
**Pull request by alexblnn**
Added one-sample calibration (func get_permuted_p_values_one_sample), sort input p-values in get_pivotal_stats, and added min_tdp function

### CODE REVIEW

**Comment by alexblnn** [comment]
get_permuted_p_values_one_sample addresses #13, sort input p-values in get_pivotal_stats addresses #17, min_tdp addresses #19

File: sanssouci/lambda_calibration.py [hunk 1]
```diff
@@ -88,6 +88,53 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
...
+    .. [1] Blanchard, G., Neuvial, P., & Roquain, E. (2020). Post hoc
+        confidence bounds on false positives using reference families.
+        Annals of Statistics, 48(3), 1281-1303.
+    """
+
+    # Init
+    n, p = X.shape
+
+    # intialise p-values
+    pval0 = np.zeros([B, p])
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:125]:
```suggestion
    pval0 = np.zeros(B, p)
```

File: sanssouci/lambda_calibration.py [hunk 2]
```diff
@@ -88,6 +88,53 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
...
+
+    for b in range(B):
+        X_flipped = (X.T * (2 * np.random.randint(-1, 1, size=n) + 1)).T
+        _, pval0[b] = stats.ttest_1samp(X_flipped, 0)
+
+    # Convert to p-values
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:131]: you can remove that one.

File: sanssouci/lambda_calibration.py [hunk 3]
```diff
@@ -88,6 +88,53 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
...
+    B : int
+        number of sign-flippings to be performed (default=100)
+
+    Returns
+    -------
+
+    pva0 : array-like of shape (B, p)
+        A numpy array of size [B,p], whose entry i,j corresponds to
+        p_{(j)}(g_i.X) with notation of the AoS 2020 paper cited below
+        (section 4.5) [1]_
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:111]: The array is sorted, right ? Please make it clear in the docstring.

File: sanssouci/post_hoc_bounds.py [hunk 4]
```diff
@@ -93,6 +93,35 @@ def min_tp(p_values, thr):
     return p_values.shape[0] - max_fp(p_values, thr)
 
 
+def min_tdp(p_values, thr):
+    """
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:97]:
```suggestion
    """Lower bound for the proportion of true discoveries in a selection
```

File: sanssouci/post_hoc_bounds.py [hunk 5]
```diff
@@ -93,6 +93,35 @@ def min_tp(p_values, thr):
...
+    Lower bound for the proportion of true discoveries in a selection
+
+    Parameters
+    -----------
+
+    p_values : 1D numpy.array
+        A 1D numpy array of p-values for the selected items
+    thr : 1D numpy.array
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:105]:
```suggestion
    threshold : 1D numpy.array
```

File: sanssouci/lambda_calibration.py [hunk 6]
```diff
@@ -118,6 +165,8 @@ def get_pivotal_stats(p0, t_inv=t_inv_linear, K=-1):
         confidence bounds on false positives using reference families.
         Annals of Statistics, 48(3), 1281-1303.
     """
+    # Sort permuted p-values
+    p0 = np.sort(p0, axis=1)
 
     # Step 3: apply template function
     tkInv_all = t_inv(p0)
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:172]:
```suggestion
    tk_inv_all = t_inv(p0)
``` use underscores for variables, modules and functions, CamelCase for classes, and never mix them !

File: sanssouci/post_hoc_bounds.py [hunk 7]
```diff
@@ -93,6 +93,35 @@ def min_tp(p_values, thr):
     return p_values.shape[0] - max_fp(p_values, thr)
 
 
+def min_tdp(p_values, thr):
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:96]:
```suggestion
def min_tdp(p_values, threshold):
```

File: sanssouci/post_hoc_bounds.py [hunk 8]
```diff
@@ -93,6 +93,35 @@ def min_tp(p_values, thr):
...
+
+    p_values : 1D numpy.array
+        A 1D numpy array of p-values for the selected items
+    thr : 1D numpy.array
+        A 1D numpy array of non-decreasing k-FWER-controlling thresholds
+
+    Returns
+    -------
+
+    ndarray or scalar :
```
Comment (bthirion) [inline sanssouci/post_hoc_bounds.py:111]: Is it possible to decide once and for all whether it's a fload or an array ? Changing signatures are a source of complexity. If you really can't avoid this, then explain in which casse it shall be a float or an array.

**Review by bthirion — COMMENTED** [review]
Thx. Some minor recommendations below.

File: sanssouci/lambda_calibration.py (hunk 3 above)
Comment (pneuvial) [inline sanssouci/lambda_calibration.py:111]:
```suggestion
    pva0 : array-like of shape (B, p)
        A numpy array of size [B,p], whose rows are sorted increasingly.
        The entry i,j corresponds to p_{(j)}(g_i.X) with notation of [1](section 4.5)_
```

**Comment by pneuvial** [comment]
Please add a unit test for the new functions

File: sanssouci/post_hoc_bounds.py (hunk 7 above)
Comment (pneuvial) [reply sanssouci/post_hoc_bounds.py:96]: Good point but for consistency with the rest of the code, it should be addressed separately. Cf #18

File: sanssouci/post_hoc_bounds.py (hunk 5 above)
Comment (pneuvial) [reply sanssouci/post_hoc_bounds.py:105]: Good point but for consistency with the rest of the code, it should be addressed separately. Cf #18

File: sanssouci/lambda_calibration.py (hunk 1 above)
Comment (alexblnn) [reply sanssouci/lambda_calibration.py:125]: Did you mean pval0 = np.zeros((B, p)) ?

File: sanssouci/lambda_calibration.py (hunk 3 above)
Comment (alexblnn) [reply sanssouci/lambda_calibration.py]: Thx, added in commit 86e6c9f885cf7410f6ccdc0741c48031b44c54c4

Comment (alexblnn) [reply sanssouci/lambda_calibration.py]: Thx, added in commit e4b7824538533c2400415a7844cfb5306f5c8149 (+ 86e6c9f885cf7410f6ccdc0741c48031b44c54c4 which corrects a wrong addition)

File: sanssouci/lambda_calibration.py (hunk 2 above)
Comment (alexblnn) [reply sanssouci/lambda_calibration.py]: thx, done in e4b7824538533c2400415a7844cfb5306f5c8149

**Review by pneuvial — APPROVED** [review]

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by bthirion** [comment]
Merged, thx.

**Pull request merged**